docs/source/dev/test.rst

```diff
@@ -0,0 +1,107 @@
+.. highlight:: cpp
+
+Writing a unit test
+==================
+
+After implementing a new functionality in Alpaka, it is recommended to test it. Indeed, having no compile time errors
+when building alpaka with any backend doesn't mean that such functionality is well implemented and behaves as expected.
+
+Unit tests are written and integrated with `Catch2`_ and they are standalone executables located in the ``test/unit`` and
+``test/integ`` paths inside the ``build`` folder. They can be built all at once by setting the ``CMake``
+argument ``-DBUILD_TESTING=ON`` and running the command ``cmake --build .``, and then run with the ``ctest`` command as
+described in the :doc:`Installation section </basic/install>`.
+
+.. _Catch2: https://github.com/catchorg/Catch2
+
+In alternative, a target test can be built by adding the flag ``--target <testname>`` to the ``cmake --build`` command and
+then executed as a normal binary file (``./<testname>``) in the corresponding folder mentioned above.
+
+Sample workflow
+-------
+
+As a sample workflow, consider the ``getWarpSize`` functionality that, given an ``alpaka device``, returns the number
+of threads that constitute a warp.
+
+First of all, the unit test should be located inside a folder under the ``test/unit`` path (namely ``dev``). Such a folder
+will contain a ``CMakeLists.txt`` file and a ``src`` folder, in which the ``.cpp`` file with the test itself will be located.
+
+A simple ``CMakeLists.txt`` file will look like:
+
+.. code-block::
+    set(_TARGET_NAME "devTest")
+
+    append_recursive_files_add_to_src_group("src/" "src/" "cpp" _FILES_SOURCE)
+
+    alpaka_add_executable(
+        ${_TARGET_NAME}
+        ${_FILES_SOURCE})
+    target_link_libraries(
+        ${_TARGET_NAME}
+        PRIVATE common)
+
+    set_target_properties(${_TARGET_NAME} PROPERTIES FOLDER "test/unit")
+
+    add_test(NAME ${_TARGET_NAME} COMMAND ${_TARGET_NAME} ${_ALPAKA_TEST_OPTIONS})
+
+where the first line sets the name of the test that ``CMake`` will build for execution. Furthermore, it is necessary to add
+the folder in which the test is located to the ``CMakeLists.txt`` file under the ``test/unit`` path:
+
+.. code-block::
+    add_subdirectory("dev/")
+
+to make sure that ``CMake`` will build the test written in the ``.cpp`` file under that directory.
+
+After these steps, it all comes to writing the *actual* test. First of all, the necessary header libraries should be
+included the ``.cpp`` file:
+
+.. code-block:: c++
+    #include <alpaka/dev/Traits.hpp>
+    #include <alpaka/test/acc/TestAccs.hpp>
+    #include <catch2/catch.hpp>
+
+The first ``include`` statement targets specifically the header in which the functionality is included; with the second one
+only accelerators corresponding to enabled backends are selected for testing, while the third one is necessary to integrate
+Catch2.
+
+One or more test cases are then introduced with the ``TEMPLATE_LIST_TEST_CASE`` macro. It takes three arguments:
```

Comment:
There might be other tests where you would want to use a different test declaration. Maybe rephrase:
```suggestion
Test cases using an alpaka accelerator are then typically introduced with the ``TEMPLATE_LIST_TEST_CASE`` macro. It takes three arguments:
```